R/shared-methods.R

```diff
@@ -130,17 +130,21 @@ DataClass <- R6::R6Class(
     process = function() {
       message_verbose(self$verbose, "Processing data")
       region_vars <- switch(self$level,
-        "1" = c("region_level_1", "level_1_region_code"),
-        "2" = c(
-          "region_level_2", "level_2_region_code",
-          "region_level_1", "level_1_region_code"
-        )
+        "1" = c("level_1_region" = self$level_1_region, 
+                "level_1_region_code" = self$level_1_region_code),
+        "2" = c("level_2_region" = self$level_2_region, 
```

Comment:
What happens if any of these are NA? i.e there are no region codes? I assume this errors. Ideally we would it not error as there are datasets where this is the case and it will make development of new datasets harder. 

Can we either insert a non-failing default or protect this code from failing when the fields are empty. I think the variable (code) should ideally not be present if its fully empty in the processed data set so that is something to consider.

Reply:
if not present in the region class this returns as NULL and doesn't get set at all - so should be OK. but i haven't tested in depth and def good to protect - will do 👍

Reply:
Amazing thank you. I would guess adding a test in `test-processing` and changing something in the stored Mexico class? (which needs to be updated to reflect the new class structure (a downside of the testing).

Reply:
confirming this works fine if not present in the region class. think i will need to update tests separately